src/test/java/JsonUtilityTest.java

```diff
@@ -12,94 +12,138 @@
 
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
+import com.google.common.collect.ImmutableMap;
 import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
 import com.google.sps.utility.JsonUtility;
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import java.lang.reflect.Type;
 import java.util.Arrays;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import javax.servlet.http.HttpServletResponse;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
-import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
 
 /** Test Json Utility methods */
 @RunWith(JUnit4.class)
 public class JsonUtilityTest {
 
-  private final String STRING_OBJECT = "HELLO WORLD";
-  private final List<Integer> LIST_OBJECT = Arrays.asList(1, 3, 5, 9, 7);
-  private final Map<String, Integer> MAP_OBJECT =
-      new HashMap<String, Integer>() {
-        {
-          put("a", 1);
-          put("c", 3);
-        }
-      };
-
-  private static HttpServletResponse response = mock(HttpServletResponse.class);
-  private static StringWriter stringWriter = new StringWriter();
-  private static PrintWriter printWriter = new PrintWriter(stringWriter);
-  private final Gson gson = new Gson();
+  private static class CustomClass {
+    String topSecret;
+
+    CustomClass(String topSecret) {
+      this.topSecret = topSecret;
+    }
+
+    public static CustomClass getInstance(String topSecret) {
+      return new CustomClass(topSecret);
+    }
 
-  @BeforeClass
-  public static void init() throws IOException {
-    // Dependency injection before tests
-    when(response.getWriter()).thenReturn(printWriter);
+    public String revealSecret() {
+      return topSecret;
+    }
   }
 
+  private static final String STRING_OBJECT = "HELLO WORLD";
+  private static final List<Integer> LIST_OBJECT = Arrays.asList(1, 3, 5, 9, 7);
+  private static final Map<String, Integer> MAP_OBJECT =
+      ImmutableMap.of(
+          "One", Integer.valueOf(1), "Two", Integer.valueOf(2), "Three", Integer.valueOf(3));
+  private static final CustomClass CUSTOM_OBJECT = CustomClass.getInstance("shhh");
+
+  private static HttpServletResponse response;
+  private static StringWriter stringWriter;
+  private static PrintWriter printWriter;
+  private final Gson gson = new Gson();
+
   @Before
-  public void flush() {
-    // Flush in case text is still in buffer
-    stringWriter.flush();
+  public void setUp() throws IOException {
+    // Mock servlet response writer
+    response = Mockito.mock(HttpServletResponse.class);
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
   }
 
   @After
-  public void clear() {
+  public void tearDown() throws IOException {
```

Comment:
If I understand correctly, this means instantiating `stringWriter` and `printWriter` in every test and to me that seems like it would add clutter. Is there another advantage to having `stringWriter` and `printWriter` as instance variables?

Reply:
JUnit instantiates a new instance of your class for every test, and runs `setUp` before the test method. You don't need to set up the objects in the test method itself - the curreset setup code would still work.

Reply:
I didn't know that, thanks for pointing this out and explaining in our standup today!